beacon-chain/blockchain/service_test.go

```diff
@@ -161,13 +158,61 @@ func SetSlotInState(service *ChainService, slot uint64) error {
 	return service.beaconDB.SaveState(bState)
 }
 
-func TestStartStop(t *testing.T) {
+func TestStartStopUninitializedChain(t *testing.T) {
+	hook := logTest.NewGlobal()
 	db := internal.SetupDB(t)
 	defer internal.TeardownDB(t, db)
 	chainService := setupBeaconChain(t, false, db)
 
 	chainService.IncomingBlockFeed()
 
+	// Test the start function.
+	chainService.Start()
+	if err := chainService.initializeBeaconChain(time.Unix(0, 0)); err != nil {
+		t.Fatalf("Error initializing: %v", err)
+	}
+
+	if err := chainService.Stop(); err != nil {
+		t.Fatalf("Unable to stop chain service: %v", err)
+	}
+
+	// The context should have been canceled.
+	if chainService.ctx.Err() == nil {
```

Comment:
Please compare to the proper error `context.Canceled` 

https://godoc.org/context#pkg-variables